R/between.R

```diff
@@ -12,7 +12,18 @@ between <- function(x,lower,upper,incbounds=TRUE) {
 }
 
 # %between% is vectorised, #534.
-"%between%" <- function(x, y) between(x, y[[1L]], y[[2L]], incbounds=TRUE)
+"%between%" <- function(x, y) {
+  if (length(y) > 2L) {
+    ysub = substitute(y)
+    warning("RHS has length() greater than 2. ",
```

Comment:
à la `if (1:5 > 1) "a"` we should indicate that part of the input is being ignored: "RHS has length > 2 and only the first and second elements will be used".

Reply:
I thought to include that, but didn't want an overlong message; you don't think that's strongly implied by the warning here?

Maybe:?

> RHS has length() greater than 2 and the rest will be ignored. [...]

Reply:
I think a problem can arise when the user says `y = c(B, B + 1)`, 'knowing' that `length(y) > 2` -- duh! It's vectorized; the warning is nothing to worry about. 

On second thoughts, maybe this should be an error. Documentation is pretty clear that `y` must be length-2, it was just never enforced. I'm trying to think of a case where only a warning is appropriate.

The suggestion (`Perhaps...`) should obviously stay: that's imo the most valuable part of this PR.

Reply:
I also was struggling to think of a use case that's not subsumed by just running `[1:2]` on whatever your input is (and even the `[1:2]` cases i couldn't think of a _real_ use case)

Reply:
@HughParsonage updated